pkg/utils/utils.go

```diff
@@ -45,3 +51,41 @@ func GetVfPciDevFromMAC(mac string) (string, error) {
 	}
 	return dev, err
 }
+
+func getRdmaDevIndexFromName(rdmaDev string) (int64, error) {
+	s := strings.Split(rdmaDev, `_`)
+	if len(s) != 2 {
+		return -1, fmt.Errorf("unexpeded RDMA device format: %s", rdmaDev)
+	}
+	idx, err := strconv.ParseInt(s[1], 0, 32)
+	if err != nil {
+		return -1, fmt.Errorf("failed to parse RDMA device index: %s, %v", rdmaDev, err)
+	}
+	return idx, nil
+}
+
+// Get the next RDMA device name for a given RDMA device prefix
+func GetNextRdmaDeviceName(prefix string, currDevs []string) (string, error) {
+	var nextDevIdx int64
+	nextDevIdx = -1
+	if len(currDevs) != 0 {
+		for _, dev := range currDevs {
+			if !strings.HasPrefix(dev, prefix) {
+				continue
+			}
+			// extract index
+			idx, err := getRdmaDevIndexFromName(dev)
+			if err != nil {
+				return "", err
+			}
+			if idx > nextDevIdx {
+				nextDevIdx = idx
+			}
+		}
+	}
+	return fmt.Sprintf("%s_%d", prefix, nextDevIdx+1), nil
+}
+
+func GetRandomRdmaDevName() string {
+	return fmt.Sprintf("rdmadev_%d", rand.Int31())[:(localIFNAMSIZ-1)]
```

Comment:
doesn't each rdma device has it own index like net device?

Reply:
yes it has, do you prefer to use `rdmadev_<rdma-dev-idx>` ?

Reply:
it ok, just remove the random